mri/reconstructors/calibrationless.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This implements calibrationless reconstruction with different proximities
+"""
+# System import
+import warnings
+
+from .base import ReconstructorBase
+from ..operators import GradAnalysis, GradSynthesis, WaveletN
+
+# Third party import
+from modopt.opt.proximity import SparseThreshold
+from modopt.opt.linear import Identity
+
+
+class CalibrationlessReconstructor(ReconstructorBase):
+    """ This class implements a regularized calibrationless reconstruction.
+    For the Analysis case, finds the solution for x of:
+        (1/2) * sum(||F x_l - y_l||^2_2, n_coils) +
+                    mu * H(W x_l)
+    For the Synthesis case, finds the solution of:
+        (1/2) * sum(||F Wt alpha_l - y_l||^2_2, n_coils) +
+                    mu * H(alpha_l)
+    Parameters
+    ----------
+    fourier_op: object of class FFT, NonCartesianFFT or Stacked3DNFFT in
+                mri.operators
+        Defines the fourier operator F in the above equation.
+    linear_op: object, (optional, default None)
+        Defines the linear sparsifying operator W. This must operate on x and
+        have 2 functions, op(x) and adj_op(coeff) which implements the
+        operator and adjoint operator. For wavelets, this can be object of
+        class WaveletN or WaveletUD2 from mri.operators .
+        If None, sym8 wavelet with nb_scale=3 is chosen.
+    regularizer_op: operator, (optional default None)
```

Comment:
it's not in the params anymore

Reply:
Im confused where to have this, can you please help me here? PLease refer to my next comment